scalyr_agent/tests/copying_manager_test.py

```diff
@@ -1022,6 +1046,66 @@ def test_start_without_active_checkpoint(self):
         self.assertEquals("Third line", lines[0])
         self.assertEquals("Fourth line", lines[1])
 
+    def test_start_with_bad_checkpoint(self):
+        controller = self.__create_test_instance()
+        previous_root_dir = os.path.dirname(self.__test_log_file)
+
+        self.__append_log_lines("First line", "Second line")
+        (request, responder_callback) = controller.wait_for_rpc()
+        lines = self.__extract_lines(request)
+
+        self.assertEquals(2, len(lines))
+        self.assertEquals("First line", lines[0])
+        self.assertEquals("Second line", lines[1])
+        controller.stop()
+
+        self.__append_log_lines("Third line", "Fourth line")
+
+        _write_bad_checkpoint_file(
+            os.path.join(self._config.agent_data_path, "active-checkpoints.json")
+        )
+        _write_bad_checkpoint_file(
+            os.path.join(self._config.agent_data_path, "checkpoints.json")
+        )
+
+        controller = self.__create_test_instance(
+            root_dir=previous_root_dir, auto_start=False
+        )
+        self._manager.start_manager()
+        (request, responder_callback) = controller.wait_for_rpc()
+        lines = self.__extract_lines(request)
+        self.assertEquals(0, len(lines))
+
+    def test_start_with_non_utf8_checkpoint(self):
+        controller = self.__create_test_instance()
```

Comment:
Add comment here explaining this is just a different form of a bad checkpoint file.